portals/portals.c

```diff
@@ -107,11 +108,13 @@ int PtlInit(void)
 
 void PtlFini(void)
 {
-	struct linked_list *current = dev_list;
-	while (current->next != NULL) {
-		struct linked_list *n = current->next;
-		free(current);
-		current = n;
+	if(dev_list != NULL) {

```

Comment:
missing space after "if"
would have preferred a separate function for freeing the swptl_dev list rather than having this code embedded into PtlFini().